### ISSUE TITLE
`http` instead of `https` when using `ssl: true`

### ISSUE THREAD

**Issue by jackoliver** [nrwl/nx#12777]
## Current Behavior
I'm noticing that when passing `ssl: true` into the serve `options`, Webpack is still displaying `http` instead of `https`. This is breaking Cypress testing without using an absolute URL because it's looking for `http` when it is actually running on `https`. I noticed in the code `get-dev-server-config.ts` in `@nrwl/webpack` that there's an `https` option respected in the serve options, but it doesn't appear in the schema and it doesn't seem to make any difference in adding it.

![image](https://user-images.githubusercontent.com/1788455/197498427-e1802906-76c3-46b4-b0aa-14866e36666e.png)

[Link to Slack thread](https://nrwlcommunity.slack.com/archives/CMFKWPU6Q/p1666602632387119)

## Expected Behavior
- Expecting `https` to be used consistently when using `ssl` or enabling the `https` option in `serve`.
- Expecting to use `https` with Cypress, without having to specify an absolute URL.

## Steps to Reproduce
- Add `ssl: true`, and generate `sslKey`, `sslCert` and add into `serve` options.
- Run an interactive instance of Cypress `e2e` and try navigating to `cy.visit('/')` in a test

### Environment
```
   Node : 16.15.1
   OS   : darwin arm64
   npm  : 8.11.0
   
   nx : 14.8.4
   @nrwl/angular : Not Found
   @nrwl/cypress : 14.8.4
   @nrwl/detox : Not Found
   @nrwl/devkit : 14.8.4
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 14.8.4
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : 14.8.4
   @nrwl/js : 14.8.4
   @nrwl/linter : 14.8.4
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : 14.8.4
   @nrwl/react-native : Not Found
   @nrwl/rollup : 14.8.4
   @nrwl/schematics : Not Found
   @nrwl/storybook : 14.8.4
   @nrwl/web : 14.8.4
   @nrwl/webpack : 14.8.4
   @nrwl/workspace : 14.8.4
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```

**Comment by s-bauer** [comment]
I have a similar issue with module federation, where the remote is served via https, but the main application is referencing it via http. I was able to fix this issue by adding the "host" options to the "@nrwl/react:module-federation-dev-server" and "@nrwl/web:file-server" executor options.

**Comment by jaysoo** [comment]
@s-bauer You can edit the remote's `serve` and `serve-static` options in `project.json`. That would allow you to skip having to specify `ssl: true` when running the servers.

This issue has been fixed as of 15.7.0.
